packages/session-replay-browser/src/session-replay.ts

```diff
@@ -0,0 +1,530 @@
+import { getAnalyticsConnector, getGlobalScope } from '@amplitude/analytics-client-common';
+import { BaseTransport, Logger } from '@amplitude/analytics-core';
+import { Logger as ILogger, ServerZone, Status } from '@amplitude/analytics-types';
+import * as IDBKeyVal from 'idb-keyval';
+import { pack, record } from 'rrweb';
+import { SessionReplayConfig } from './config';
+import {
+  BLOCK_CLASS,
+  DEFAULT_SESSION_REPLAY_PROPERTY,
+  MASK_TEXT_CLASS,
+  MAX_EVENT_LIST_SIZE_IN_BYTES,
+  MAX_IDB_STORAGE_LENGTH,
+  MAX_INTERVAL,
+  MIN_INTERVAL,
+  SESSION_REPLAY_EU_URL as SESSION_REPLAY_EU_SERVER_URL,
+  SESSION_REPLAY_SERVER_URL,
+  STORAGE_PREFIX,
+  defaultSessionStore,
+} from './constants';
+import { isSessionInSample, maskInputFn } from './helpers';
+import {
+  MAX_RETRIES_EXCEEDED_MESSAGE,
+  MISSING_API_KEY_MESSAGE,
+  MISSING_DEVICE_ID_MESSAGE,
+  STORAGE_FAILURE,
+  UNEXPECTED_ERROR_MESSAGE,
+  getSuccessMessage,
+} from './messages';
+import {
+  AmplitudeSessionReplay,
+  Events,
+  IDBStore,
+  IDBStoreSession,
+  SessionReplayConfig as ISessionReplayConfig,
+  RecordingStatus,
+  SessionReplayContext,
+  SessionReplayOptions,
+} from './typings/session-replay';
+
+export class SessionReplay implements AmplitudeSessionReplay {
+  name = '@amplitude/session-replay-browser';
+  config: ISessionReplayConfig | undefined;
+  loggerProvider: ILogger;
+  storageKey = '';
+  retryTimeout = 1000;
+  events: Events = [];
+  currentSequenceId = 0;
+  private scheduled: ReturnType<typeof setTimeout> | null = null;
+  queue: SessionReplayContext[] = [];
+  stopRecordingEvents: ReturnType<typeof record> | null = null;
+  maxPersistedEventsSize = MAX_EVENT_LIST_SIZE_IN_BYTES;
+  interval = MIN_INTERVAL;
+  timeAtLastSend: number | null = null;
+
+  constructor() {
+    this.loggerProvider = new Logger();
+  }
+
+  async init(apiKey: string, options: SessionReplayOptions) {
```

Comment:
do we need to handle the case when init is called multiple times? That was a way to disable/reenable with browser sdk.

Reply:
oh just saw, there is teardown. should we enforce the need to teardown before re-initializing?

Reply:
No it's actually not necessary, I tested calling init multiple times without teardown, and there are no negative repercussions. Teardown does 2 things, 1) stops recording events and 2) removes event listeners. 

If teardown is not called before init is called a subsequent time, recording has not been stopped, which means that sessionReplay.stopRecordingEvents is defined. We return early [here](https://github.com/amplitude/Amplitude-TypeScript/blob/78ad90a58053d4971bf54231bcf96b42ff90b6b1/packages/session-replay-browser/src/session-replay.ts#L143-L143) and prevent a new `record` call from occurring is sessionReplay.stopRecordingEvents is already defined.

Additionally, there are no additional event listeners added because addEventListener is called with the same reference as it has already been called with, which does not result in a new listener (https://stackoverflow.com/a/26146974).

So teardown is really just a utility that can be called if the customer wants to stop recording altogether for a while, it doesn't need to come before init.